example/lib/pages/login_page.dart

```diff
@@ -58,27 +58,39 @@ class LoginPage extends HookWidget {
                   }
                 },
               ),
-              
               ElevatedButton(
                 child: const Text('Create Random Wallet'),
                 onPressed: () async {
                   try {
                     print('Creating a random wallet...');
-                    var signer = EthPrivateKey.createRandom(Random.secure()).asSigner();
-                    print('Wallet created with address: ${await signer.address}');
+                    var signer =
+                        EthPrivateKey.createRandom(Random.secure()).asSigner();
+                    print(
+                        'Wallet created with address: ${await signer.address}');
                     print('Authorizing the wallet...');
                     bool isAuthorized = await session.authorize(signer);
                     print('Wallet authorized: $isAuthorized');
-                    print(context);
-                    print('Session: $session');
-                    print('Context: $context');
+                    if (isAuthorized) {
+                      await session.loadSaved();
+                    }
+                    // print(context);
+                    // print('Session: $session');
+                    // print('Context: $context');
+                    // Future.delayed(const Duration(seconds: 10), () {
+                    if (!context.mounted) return;
+                    print('Navigating to home2...');
                     context.goNamed('home');
+                    // print('Navigation to home completed.');
+                    // });
                     print('Navigation to home completed.');
                   } catch (e) {
-                    print('Errorf: $e');
+                    print('Error within Create Random Wallet: $e');
+                    if (e.toString().contains('already initialized')) {
+                      print('Already initialized, so just navigating to home');
+                      context.goNamed('home');
+                    }
```

Comment:
In case we get any exceptions and we had previously initialized the wallet, we can add here this logic to still be able to login.

I would check this with your team though.